a10_octavia/controller/worker/tasks/glm_tasks.py

```diff
@@ -0,0 +1,253 @@
+            secondary_dns = CONF.glm_license.secondary_dns
+
+        if flavor:
+            dns_flavor = flavor.get('dns')
+            if dns_flavor and dns_flavor.get('primary-dns'):
+                primary_dns = dns_flavor.get('primary-dns')
+            if dns_flavor and dns_flavor.get('secondary-dns'):
+                secondary_dns = dns_flavor.get('secondary-dns')
+
+        return primary_dns, secondary_dns
+
+    @axapi_client_decorator
+    def execute(self, vthunder, flavor=None):
+        if not vthunder:
+            LOG.warning("No vthunder therefore dns cannot be assigned.")
+            return None
+
+        flavor = flavor if flavor else {}
+        primary_dns, secondary_dns = self._get_dns_nameservers(vthunder, flavor)
+        if not primary_dns and secondary_dns:
+            LOG.error("A secondary DNS with IP %s was specified without a primary DNS",
+                      primary_dns)
+            raise a10_ex.PrimaryDNSMissing(secondary_dns)
+        elif not primary_dns and not secondary_dns:
+            return None
+
+        try:
+            self.axapi_client.dns.set(primary_dns, secondary_dns)
+        except acos_errors.ACOSException as e:
+            LOG.error("Could not set DNS configuration for amphora %s",
+                      vthunder.amphora_id)
+            raise e
+
+    @axapi_client_decorator_for_revert
+    def revert(self, vthunder, flavor=None, *args, **kwargs):
+        flavor = flavor if flavor else {}
+        primary_dns, secondary_dns = self._get_dns_nameservers(vthunder, flavor)
+        if not primary_dns and not secondary_dns:
+            return None
+
+        try:
+            self.axapi_client.dns.delete(primary_dns, secondary_dns)
+        except req_exceptions.ConnectionError:
+            LOG.exception("Failed to connect A10 Thunder device: %s", vthunder.ip_address)
+
+
+class ActivateFlexpoolLicense(task.Task):
+
+    def _build_configuration(self, flavor={}):
+        glm_config = dict(CONF.glm_license)
+        glm_config.update(utils.dash_to_underscore(flavor.get('glm', {})))
+
+        config_payload = {
+            "port": glm_config.get('port'),
+            "burst": glm_config.get('burst'),
+            "token": glm_config.get('flexpool_token'),
+            "interval": glm_config.get('interval'),
+            "enable_requests": glm_config.get('enable_requests'),
+            "allocate_bandwidth": glm_config.get('allocate_bandwidth'),
+        }
+        return config_payload
+
+    @axapi_client_decorator
+    def execute(self, vthunder, amphora, flavor=None):
+        if not vthunder:
+            LOG.warning("No vthunder therefore licensing cannot occur.")
+            return None
+
+        flavor = flavor if flavor else {}
+        amp_mgmt_net = CONF.a10_controller_worker.amp_mgmt_network
+        amp_boot_nets = CONF.a10_controller_worker.amp_boot_network_list
+        config_payload = self._build_configuration(flavor)
+        hostname = "amphora-" + amphora.id
```

Comment:
Can we return if token not in config_payload?
Otherwise a10-octavia will always failed if no glm configured.